compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -1096,9 +1156,11 @@ impl<'context> Elaborator<'context> {
         lhs_type: &Type,
         op: &HirBinaryOp,
         rhs_type: &Type,
-        span: Span,
+        location: Location,
     ) -> bool {
-        self.unify(lhs_type, rhs_type, || TypeCheckError::TypeMismatchWithSource {
+        let span = location.span;
+
+        self.unify(lhs_type, rhs_type, location.file, || TypeCheckError::TypeMismatchWithSource {
```

Comment:
Definitely something to leave for a follow-up issue, but I think that we should try to move all these `file`'s to the error enums so that `push_err` and `unify` don't need to keep track of them.

Reply:
Definitely! It's something I wanted to do, and I actually considered doing as part of this PR, but I found that not doing it led to a smaller PR. I captured https://github.com/noir-lang/noir/issues/7452 for this.

Reply:
Also to clarify: I started doing that but before that I had almost one thousand errors and that change brought it to 1K+ so the number of errors was unknown (at least in VSCode) and it demotivated me a bit to continue without seeing the number of errors going down. Then I realized requiring a file in `push_err` would work "the same way" so I took that approach.